public/javascript/getCatPicture.js

```diff
@@ -0,0 +1,51 @@
+/**
+ * @file getCatPicture.js
+ * Handler for getting cat pics.
+ *
+ */
+'use strict'
+ 
+var request = require('request');
+var fs = require('fs');
+
+var type = 'png';
+
+module.exports = function getCatPicture(api, threadID, body) {
+    if (body == '@cat gif') {
+        type = 'gif';
+    }
+    var link;
+    var API_KEY = process.env.CAT_API_KEY;
+    if (API_KEY != undefined) {
+        API_KEY = '&api-key=' + API_KEY;
+    }
```

Comment:
Oh that's my sketchy code. [If there's no key the API still works](http://thecatapi.com/docs.html), but you only have access to a limited amount of cat pics.